### PR TITLE
cleaned up 3 examples [MEP12]

### PR DESCRIPTION
**Pull request by aneda**
The following three examples:
-- boxplot_color_demo.py
-- boxplot_vs_violin_demo.py
-- bachelors_degrees_by_gender.py

### CODE REVIEW

**Comment by tacaswell** [comment]
:+1: will merge after CI passes.

File: examples/showcase/bachelors_degrees_by_gender.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
+"""Timeline graph comparison with aligned colored annotation.
+
+This example shows lines tracking change over time.  It features dashed
+lines at fixed increments on the Y axis.
```
Comment (WeatherGod) [inline examples/showcase/bachelors_degrees_by_gender.py]: being a bit pedantic here, but this paragraph could be used to describe just about any plot. I would just leave it out, or add a some info that is more specific to this example. In particular, with MEP12, we are looking for descriptions that not only describe the figure, but also describe what feature of matplotlib is being showcased.

Comment (tacaswell) [inline examples/showcase/bachelors_degrees_by_gender.py]: This one is a particularly flexibility testing example.

Comment (discardthree) [inline examples/showcase/bachelors_degrees_by_gender.py]: What about something like...
"""
A graph of multiple time series which demonstrates extensive custom
styling of plot frame, tick lines and labels, and line graph properties.

Also demonstrates the custom placement of text labels along the right edge
as an alternative to a conventional legend.
"""

Comment (WeatherGod) [inline examples/showcase/bachelors_degrees_by_gender.py]: Perfect!

**Comment by fariza** [comment]
@aneda if you don't have time I don't mind taking over this PR to get it merged

**Comment by aneda** [comment]
@fariza go ahead!  I had forgotten about this and I'd thought it was done

File: examples/statistics/boxplot_color_demo.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Box plots with custom fill colors
+"""Box plots with custom fill colors."""
```
Comment (NelleV) [inline examples/statistics/boxplot_color_demo.py]: It would be worth having a small description here as well.

Comment (fariza) [inline examples/statistics/boxplot_color_demo.py]: @NelleV Any suggestions?

**Comment by tacaswell** [comment]
Sorry for my noise on this :backs slowly away:

File: examples/statistics/boxplot_color_demo.py (hunk 2 above)
Comment (NelleV) [inline examples/statistics/boxplot_color_demo.py]: You are making me work :)

Something like: 
"This plot illustrates how to create two types of boxplots (rectangular and notch), and how to fill them with custom colors by accessing some properties of the artists of the boxplots."
If we'd go with this description, I'd replace the title with something shorter: "Colored box plots"

But with this description, I'd remove the horizontal grid lines and replace the `setp` with `ax.set_xticklabels`.

**Comment by NelleV** [comment]
@tacaswell Are we changing the title when we approve the PR or not? The PR review guidelines say so, but you undid your title change :)

File: examples/statistics/boxplot_color_demo.py (hunk 2 above)
Comment (fariza) [inline examples/statistics/boxplot_color_demo.py]: I'm not convinced, @WeatherGod  what do you think?

**Comment by fariza** [comment]
@tacaswell @WeatherGod I just pushed the replacement PR to finish this review process,

**Pull request closed without merging**